### PR TITLE
implemented 517

### PR DESCRIPTION
**Pull request by ineiti**
Waiting for Travis...

### CODE REVIEW

File: crypto/hash.go [hunk 1]
```diff
@@ -108,6 +113,32 @@ func HashArgs(hash h.Hash, args ...encoding.BinaryMarshaler) ([]byte, error) {
...
 	return HashArgs(suite.Hash(), args...)
 }
+
+// ConvertToBinaryMarshaler takes a slice of interfaces and returns
+// a slice of BinaryMarshalers.
+func ConvertToBinaryMarshaler(args ...interface{}) ([]encoding.BinaryMarshaler, error) {
+	var ret []encoding.BinaryMarshaler
+	for _, a := range args {
+		refl := reflect.ValueOf(a)
+		if refl.Kind() == reflect.Slice {
```
Comment (nikkolasg) [inline crypto/hash.go]: Should have a if for Array also no ?

Comment (ineiti) [inline crypto/hash.go]: Hmm - probably yes.

File: crypto/hash.go [hunk 2]
```diff
@@ -108,6 +113,32 @@ func HashArgs(hash h.Hash, args ...encoding.BinaryMarshaler) ([]byte, error) {
...
 
 // HashArgsSuite makes a new hash from the suite and calls HashArgs
-func HashArgsSuite(suite abstract.Suite, args ...encoding.BinaryMarshaler) ([]byte, error) {
+func HashArgsSuite(suite abstract.Suite, args ...interface{}) ([]byte, error) {
 	return HashArgs(suite.Hash(), args...)
 }
+
+// ConvertToBinaryMarshaler takes a slice of interfaces and returns
+// a slice of BinaryMarshalers.
+func ConvertToBinaryMarshaler(args ...interface{}) ([]encoding.BinaryMarshaler, error) {
```
Comment (nikkolasg) [inline crypto/hash.go]: Wondering if we should let that exported or not. I don't see any reasons why it should be exported for the moment.

Comment (ineiti) [inline crypto/hash.go]: for the moment it's only exported so that I can use it in the `crypto_test`...

Comment (nikkolasg) [inline crypto/hash.go]: Damn, good thing you talk about it. Can we go back to `package crypto` in test ? It's a *unit test* here, I see no reason to put it externally.

Comment (ineiti) [inline crypto/hash.go]: OK - I'll do that first thing tomorrow

**Comment by ineiti** [comment]
Changed the tests to be unit-tests.

**Comment by nikkolasg** [comment]
OK to merge.

**Pull request merged**